cql3/expr/expression.cc

```diff
@@ -818,7 +818,7 @@ bool has_supporting_index(
         const secondary_index::secondary_index_manager& index_manager,
         allow_local_index allow_local) {
     const auto indexes = index_manager.list_indexes();
-    const auto support = std::bind(is_supported_by, expr, std::placeholders::_1);
+    const auto support = std::bind(is_supported_by, std::ref(expr), std::placeholders::_1);
```

Comment:
Looks good, I'll commit. But personally, I would have done this with lambdas, not std::bind. With lambdas it's much clearer what is being captured and how.